apps/stats/stats-1.py

```diff
@@ -743,13 +741,7 @@ def __init__( self, bytes ) :
 	@classmethod
 	def maxRSS( cls ) :
 
-		if sys.platform == "darwin" :
-			return cls( resource.getrusage( resource.RUSAGE_SELF ).ru_maxrss )
-		elif sys.platform == "win32" :
-			result = subprocess.check_output( ["wmic", "process", "where", "processid={}".format(os.getpid()), "get", "PeakWorkingSetSize"] )
-			return cls( int( result.split()[1] ) * 1024 )
-		else :
-			return cls( resource.getrusage( resource.RUSAGE_SELF ).ru_maxrss * 1024 )
+		return cls( psutil.Process().memory_info().rss )
```

Comment:
As you ponder in the commit message, this doesn't seem an ideal replacement as it gives us a sample in time rather than peak memory. From a little searching there doesn't appear to be a cross-platform way of accessing peak memory from `psutil`, but we could potentially replace the existing Windows-specific `subprocess.check_output( ["wmic", ...` with the also Windows-specific `psutil.Process().memory_info().peak_wset`.

https://psutil.readthedocs.io/en/latest/#psutil.Process.memory_info

Reply:
Cool. I've dropped that commit and done as you suggest for Windows in https://github.com/GafferHQ/gaffer/pull/5577/commits/ef57c631e696b4054099fd7bff90afffc5814490. I tested on Windows and `memory().rss` was giving us smaller numbers than the previous approach, whereas `memory_info().peak_wset` matches very closely, and avoids the subprocess we were using before.